### ISSUE TITLE
Qwen3-VL GRPO训练rollout卡住超时

### ISSUE THREAD

**Issue by LuckyDoggy** [modelscope/ms-swift#6274]
使用主分支代码，Qwen3-VL-8B-Instruct模型进行视频数据的GRPO训练，会在训练几个step后rollout卡住超时

rollout脚本
```shell
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True' \
VIDEO_MIN_TOKEN_NUM=48 \
VIDEO_MAX_TOKEN_NUM=48 \
FPS_MAX_FRAMES=360 \
DECORD_EOF_RETRY_MAX=20480 \
CUDA_VISIBLE_DEVICES=4,5,6,7 \
swift rollout \
  --model /workspace/Qwen3-VL-8B-Instruct/ \
  --vllm_tensor_parallel_size 1 \
  --vllm_data_parallel_size 4 \
  --temperature 1.0 \
  --vllm_gpu_memory_utilization 0.6 \
  --vllm_max_model_len 17536
```

训练脚本
```shell
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True' \
VIDEO_MIN_TOKEN_NUM=48 \
VIDEO_MAX_TOKEN_NUM=48 \
FPS_MAX_FRAMES=360 \
MASTER_PORT=29501 \
DECORD_EOF_RETRY_MAX=20480 \
CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model /workspace/Qwen3-VL-8B-Instruct/ \
    --add_version false \
    --load_from_cache_file true \
    --reward_funcs Task_reward \
    --reward_weights 1.0 \
    --train_type full \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8001 \
    --vllm_gpu_memory_utilization 0.6 \
    --vllm_max_model_len 17536 \
    --torch_dtype bfloat16 \
    --dataset ${dataset1} \
    --max_completion_length 256 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps 8 \
    --split_dataset_ratio 0 \
    --save_steps 400 \
    --save_total_limit 500 \
    --logging_steps 1 \
    --max_length 17536 \
    --output_dir ${output_dir} \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 1.0 \
    --top_p 0.9 \
    --top_k 50 \
    --num_iterations 1 \
    --deepspeed zero3 \
    --attn_impl flash_attn \
    --log_completions true \
    --overlong_filter True \
    --use_liger_kernel True \
    --external_plugins /workspace/plugin.py
```

rollout报错截图

<img width="1555" height="845" alt="Image" src="https://github.com/user-attachments/assets/52c842f2-a906-47ef-bf22-0deed45b612b" />

**Comment by LuckyDoggy** [comment]
硬件配置： 8 * H100 80G
环境如下：
absl-py                           2.3.1
accelerate                        1.11.0
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.1
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.11.0
astor                             0.8.1
asttokens                         3.0.0
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.4.0
av                                16.0.1
binpacking                        1.5.2
blake3                            1.0.8
boto3                             1.40.55
botocore                          1.40.55
Brotli                            1.1.0
cachetools                        6.2.1
cbor2                             5.7.0
certifi                           2025.10.5
cffi                              2.0.0
charset-normalizer                3.4.4
click                             8.2.1
cloudpickle                       3.1.1
colorama                          0.4.6
colorlog                          6.10.1
compressed-tensors                0.11.0
contourpy                         1.3.2
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      46.0.3
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.6.0
decorator                         5.2.1
decord                            0.6.0
deepspeed                         0.17.5
depyf                             0.19.0
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
docstring_parser                  0.17.0
dotenv                            0.9.9
einops                            0.8.1
email-validator                   2.3.0
et_xmlfile                        2.0.0
evalscope                         1.1.0
evaluate                          0.4.6
exceptiongroup                    1.3.0
executing                         2.2.1
fastapi                           0.119.1
fastapi-cli                       0.0.14
fastapi-cloud-cli                 0.3.1
fastrlock                         0.8.3
ffmpy                             0.6.3
filelock                          3.20.0
fire                              0.7.1
flash_attn                        2.7.4.post1
fonttools                         4.60.1
frozendict                        2.4.6
frozenlist                        1.8.0
fsspec                            2025.3.0
func_timeout                      4.3.5
future                            1.0.0
fuzzywuzzy                        0.18.0
gguf                              0.17.1
google-auth                       2.41.1
google-genai                      1.45.0
gradio                            5.49.1
gradio_client                     1.13.3
groovy                            0.1.2
grpcio                            1.75.1
h11                               0.16.0
h5py                              3.15.1
hf-xet                            1.1.10
hjson                             3.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.35.3
human-eval                        1.0.3
idna                              3.11
imageio                           2.37.0
immutabledict                     4.2.2
importlib_metadata                8.7.0
interegular                       0.3.3
ipdb                              0.13.13
ipython                           8.37.0
jedi                              0.19.2
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.11.1
jmespath                          0.10.0
joblib                            1.5.2
json_repair                       0.52.2
json5                             0.12.1
jsonlines                         4.0.0
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
kiwisolver                        1.4.9
langdetect                        1.0.9
lark                              1.2.2
latex2sympy2_extended             1.0.6
Levenshtein                       0.27.1
liger_kernel                      0.6.2
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
lxml                              6.0.2
Markdown                          3.9
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
math-verify                       0.5.2
matplotlib                        3.10.7
matplotlib-inline                 0.1.7
mdurl                             0.1.2
mistral_common                    1.8.5
mmengine-lite                     0.10.7
modelscope                        1.31.0
mpmath                            1.3.0
ms-opencompass                    0.1.6
ms_swift                          3.10.0.dev0   /workspace/ms-swift
ms-vlmeval                        0.0.18
msgpack                           1.1.2
msgspec                           0.19.0
multidict                         6.7.0
multiprocess                      0.70.16
networkx                          3.4.2
ninja                             1.13.0
nltk                              3.9.2
numba                             0.61.2
numpy                             1.26.4
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-ml-py                      13.580.82
nvidia-nccl-cu12                  2.27.3
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvtx-cu12                  12.8.90
openai                            2.6.0
openai-harmony                    0.0.4
OpenCC                            1.1.9
opencv-python                     4.11.0.86
opencv-python-headless            4.12.0.88
openpyxl                          3.1.5
orjson                            3.11.3
oss2                              2.19.1
outlines_core                     0.2.11
overrides                         7.7.0
packaging                         25.0
pandas                            2.3.3
parso                             0.8.5
partial-json-parser               0.2.1.1.post6
peft                              0.17.1
pexpect                           4.9.0
pillow                            11.3.0
pip                               25.2
platformdirs                      4.5.0
portalocker                       3.2.0
prettytable                       3.16.0
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
prompt_toolkit                    3.0.52
propcache                         0.4.1
protobuf                          6.33.0
psutil                            7.1.1
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
pyarrow                           21.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pydantic                          2.11.10
pydantic_core                     2.33.2
pydantic-extra-types              2.10.6
pydub                             0.25.1
pyecharts                         2.0.9
Pygments                          2.19.2
pyparsing                         3.2.5
pypinyin                          0.55.0
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.1
python-json-logger                4.0.0
python-Levenshtein                0.27.1
python-multipart                  0.0.20
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-vl-utils                     0.0.14
rank-bm25                         0.2.2
RapidFuzz                         3.14.1
ray                               2.50.1
referencing                       0.37.0
regex                             2025.9.18
requests                          2.32.5
rich                              13.9.4
rich-toolkit                      0.15.1
rignore                           0.7.1
rouge                             1.0.1
rouge-chinese                     1.0.3
rouge_score                       0.1.2
rpds-py                           0.27.1
rsa                               4.9.1
ruff                              0.14.1
s3transfer                        0.14.0
sacrebleu                         2.5.1
safehttpx                         0.1.6
safetensors                       0.6.2
scikit-learn                      1.7.2
scipy                             1.15.3
seaborn                           0.13.2
semantic-version                  2.10.0
sentence-transformers             5.1.1
sentencepiece                     0.2.1
sentry-sdk                        2.42.1
setproctitle                      1.3.7
setuptools                        80.9.0
shellingham                       1.5.4
simplejson                        3.20.2
six                               1.17.0
sniffio                           1.3.1
sortedcontainers                  2.4.0
soundfile                         0.13.1
soxr                              1.0.0
stack-data                        0.6.3
starlette                         0.48.0
sty                               1.0.6
swanlab                           0.6.12
sympy                             1.14.0
tabulate                          0.9.0
tenacity                          9.1.2
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
termcolor                         3.1.0
threadpoolctl                     3.6.0
tiktoken                          0.12.0
timeout-decorator                 0.5.0
timm                              1.0.20
tokenizers                        0.22.1
tomli                             2.3.0
tomlkit                           0.13.3
torch                             2.8.0
torchaudio                        2.8.0
torchvision                       0.23.0
tqdm                              4.67.1
traitlets                         5.14.3
transformers                      4.57.1
transformers-stream-generator     0.0.5
triton                            3.4.0
trl                               0.20.0
typer                             0.20.0
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.38.0
uvloop                            0.22.1
validators                        0.35.0
vllm                              0.11.0
watchfiles                        1.1.1
wcwidth                           0.2.14
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
word2number                       1.1
wrapt                             2.0.0
xformers                          0.0.32.post1
xgrammar                          0.1.25
xlsxwriter                        3.2.9
xxhash                            3.6.0
yapf                              0.43.0
yarl                              1.22.0
zipp                              3.23.0
zstandard                         0.25.0

**Comment by jhl-Det** [comment]
same issue here

**Comment by jhl-Det** [comment]
<img width="715" height="82" alt="Image" src="https://github.com/user-attachments/assets/b5028edc-1ac1-434f-9d70-5da0c6fd09a1" />

**Comment by hjh0119** [comment]
Check whether the freeze is caused by abnormal data (e.g., videos that are unusually long or fail to load). I tested with the same short video clip and could not reproduce this issue.

**Comment by alanshaoTT** [comment]
请问最终是如何解决的
